clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/provider/view/AppEngineProviderUtils.groovy

```diff
@@ -35,14 +35,18 @@ class AppEngineProviderUtils {
   }
 
   static AppEngineInstance instanceFromCacheData(ObjectMapper objectMapper, CacheData instanceData) {
-    def instance = objectMapper.convertValue(instanceData.attributes.instance, AppEngineInstance)
-    def loadBalancers = instanceData.relationships[Keys.Namespace.LOAD_BALANCERS.ns]?.collect { Keys.parse(it).name } ?: []
-    instance.loadBalancers = loadBalancers
+    if (!instanceData) {
+      return null
+    } else {
+      def instance = objectMapper.convertValue(instanceData.attributes.instance, AppEngineInstance)
+      def loadBalancers = instanceData.relationships[Keys.Namespace.LOAD_BALANCERS.ns]?.collect { Keys.parse(it).name } ?: []
+      instance.loadBalancers = loadBalancers
 
-    def serverGroup = instanceData.relationships[Keys.Namespace.SERVER_GROUPS.ns]?.collect { Keys.parse(it).name }?.first()
-    instance.serverGroup = serverGroup
+      def serverGroup = instanceData.relationships[Keys.Namespace.SERVER_GROUPS.ns]?.collect { Keys.parse(it).name }?.first()
```

Comment:
Does appengine allow you to have instances that aren't managed by a server group?